### ISSUE TITLE
CANONICAL.HOSTNAME undefined function

### ISSUE THREAD

**Issue by rmkaplan** [Interlisp/medley#297]
Error when doing Hardcopy off of background menu in a Tedit window.

In a full.sysout that I just made from the current master, date March 22 19:07

**Comment by Anzus** [comment]
If memory serves, that’s in the PUP code. Possibly it’s trying to map a filename to a host?

**Comment by Anzus** [comment]
Interlisp-D the operating system went to great lengths to make file access seamless across network devices and protocols, quite frankly better than most other operating systems of that era. And now that's coming back to bite us as we consider it more as an application/environment layered on an underlying operating system.On Mar 22, 2021 10:17 PM, Arun Welch ***@***.***> wrote:If memory serves, that’s in the PUP code. Possibly it’s trying to map a filename to a host? On Mar 22, 2021, at 10:13 PM, rmkaplan ***@***.***> wrote:Error when doing Hardcopy off of background menu in a Tedit window.In a full.sysout that I just made from the current master, date March 22 19:07—You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by nbriggs** [comment]
Defined in PUP, used in
```
sources/DPUPFTP
sources/HARDCOPY
sources/INTERPRESS
sources/LEAF
sources/PASSWORDS
lispusers/DIRGRAPHER
lispusers/DPUPFTPPATCH
lispusers/MATHSERVER
lispusers/SETDEFAULTPRINTER
```

**Comment by rmkaplan** [comment]
I looked in my pre-repository directory and indeed it is in PUP. It seems to be canonicalizing NS, IP, and PUP hostnames.

This came up while testing the coercion of my old Bravo files to tedit, with the user.cm.  It all looks pretty good on the display, once I manually adjust the margins.

But hardcopy is a problem for something in the pre XCCS Bravo days.  It has characters that display as Greek but are coded in the file as Ascii equivalents (e.g. b for beta) in the Greek font HIPPO.  We have a display font for HIPPO, but no corresponding postscript font metrics.

It doesn’t look like Tedit supports a formatted find/replace, sadly.

**Comment by masinter** [comment]
modularity is lost to the layers of patching done with different assumptions.
There were lots of people involved. Most of whom weren't taking the time to figure out where a patch "should" go.
In this case, though, the whole idea of a 'canonical.hostname' doesn't really fit.
`MOVD?(IDENTITY CANONICAL.HOSTNAME)` 
I tried to reproduce the problem and couldn't

**Comment by nbriggs** [comment]
And any other functions that are in PUP and used elsewhere?   How about we put it back in the release, and do the analysis of what can be removed or moved beforehand.

**Comment by Anzus** [comment]
Is it possible to automate this via Masterscope? Sort of a WHO USES any of the symbols in the filecoms across the rest of the system? Nick, is that how you came up with your list or did you use grep against the source files?

**Comment by masinter** [comment]
I've been using
```
EDITMODE(TELETYPE)
(LENGTH (SETQ ALLCONTENT (FOR X IN SYSFILES COLLECT (CONS X (READFILE(FINDFILE X]
(SETQ MAXDEPTH 3000) (SETQ MAXLOOP 20000)
ED(ALLCONTENT)
```
Masterscope has gotten confused about functions vs macros, Whar I'm puzzled by is why `MACRO-FN` isn't on` MSMACROPROPS

**Comment by nbriggs** [comment]
It was grep.  What we need is "who calls any on X" or "who uses any on X freely"  -- and to do that before any file gets removed from the image.   But then the documentation is incorrect (CANONICAL.HOSTNAME is documented in the I/O volume) too.  I don't see that arbitrarily taking pieces of the system out is improving it in any meaningful way -- and it is certainly making work to go back and figure out what got broken, and I don't see anyone stepping up to fix the documentation of each function/feature that's being removed.

**Comment by nbriggs** [comment]
p.s., when the host-based network code is working we'll still want to do a CANONICAL.HOSTNAME.

**Comment by masinter** [comment]
PUP and XNS had code in patches directory to DELETE the code for PUP and XNS (the XNS code deletion was buggy, which is why you’d see \10MB.GATEWAY.LISTENER in full.venuesysout.

**Comment by nbriggs** [comment]
LEAF and PUP and the XNS files were (are) still in the Venue full.sysout that I run.
If there's no ethernet I can believe that somebody's patches tried to have it not listen on the network interfaces, but all of the infrastructure is still there.

**Comment by masinter** [comment]
fixed by #298
